src/ActionsTests/Actions/ScreenShotActionUnitTests.cs

```diff
@@ -29,127 +29,103 @@ private void SetBoundingRectangle(A11yElement element, Rectangle b)
             element.Properties.Add(typeId, new A11yProperty(typeId, data));
         }
 
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            ScreenShotAction.GetDataManager = () => DataManager.GetDefaultInstance();
```

Comment:
Code coverage shows that the call to GetDefaultInstance is never being called here. That would suggest that this line can be removed

Reply:
But don't you agree that the pattern is good practice? I think it's good to reset such values to their defaults in case someone comes along behind us and writes a new unit test, then gets unexpected results that would be hard to track down.

Reply:
I agree with that principle, but I think if that's the intended purpose it might be better to do this in either a `[TestCleanup]` method or a `Dispose()` of a using-based helper to temporarily set it to something different (see the `CaptureActionTestHookOverrides` class inside `CaptureActionUnitTests` for an example of this)

Reply:
One of the benefits of the fakes was that we never accidentally fell through to any underlying classes. This pattern seems to discard that benefit, so I'm not really a fan. Setting these on a per-test basis feels like the better approach. If we want to make the tests easy to debug, the TestInitialize code would call an Assert.Fail with an error message to tell the code writer that they forgot to set the test override.

Reply:
@dbjorge I have updated the code to use [TestCleanup], which is the right attribute to use since it ensures the DataManager will be reset after it's potentially been changed and not before it's potentially used. So thanks for that.

Before I began, I studied your solution in CaptureActionTestHookOverrides. If this were production code, I would have considered using either that technique or adding a method with a callback to ensure the static members were reset to their default values. However, this is not production code and the tests provide a mechanism for exactly this purpose, which I used because it's simpler and conventional in unit tests.

@DaveTryon Let's don't lose sight of what we're trying to do here and the environment (singletons) in which we find ourselves working. None of us likes the static injection. The options are to

1. Continue using Fakes
2. Use a RAII technique as Dan did
3. Use a callback mechanism which is essentially like RAII except calling a method instead of creating an object
4. Using [TestCleanup]
5. Rewrite the code to use interfaces and mocks

Using Fakes is obviously not desireable. As I explained above, I find the RAII techniques to be unnecessary given that unit tests have both a mechanism and a convention to deal with this type of problem. And I don't think we want to invest in a rewrite of the code. Ergo, I chose the solution that I did.